frontend/src/stores/notificationStore.ts

```
@@ -0,0 +1,373 @@
+    // Clean up existing transport
+    this.disconnect();
+
+    const token = options?.token || localStorage.getItem('authToken') || '';
+    const apiUrl =
+      options?.apiUrl ||
+      import.meta.env.VITE_API_BASE_URL ||
+      'http://localhost:3001';
+
+    switch (transportType) {
+      case 'websocket':
+        const socketUrl = `${apiUrl.replace('/api', '')}/notifications`;
```

Biome (2.4.13)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)